test/test_core.py

```diff
@@ -6016,6 +6016,7 @@ def test_fs_js_api(self):
     self.do_runf(test_file('fs/test_fs_js_api.c'), 'success')
 
   def test_fs_write(self):
+    self.set_setting("FORCE_FILESYSTEM")
```

Comment:
Maybe best to only do this in wasmfs mode,
```suggestion
    if self.get_setting('WASMFS'):
      self.set_setting("FORCE_FILESYSTEM")
```
Though, I can see an argument that uniformity is better, so I'm not sure.

@sbc100 what do you think?

Reply:
Adding the if statement for WASMFS causes the test to fail for the legacy API, since I think we need `FORCE_FILESYSTEM` for the legacy JS API to see all the methods. 

Could there be an issue with the way I'm running the tests that means Emscripten doesn't automatically include the filesystem?

Reply:
I'm surprised the old FS fails without `FORCE_FILESYSTEM` - what error do you get there?

The old FS will include the `FS` object when any filesystem usage is present, automatically, so it should work. There are some exceptions but I can't see anything relevant here.

Reply:
Here is the error: 
```
TypeError: FS.writeFile is not a function
    at 65696 (/home/ubuntu/Documents/emsdk/emscripten/out/test/test_fs_js_api.js:1023:21)
```
I got the error both locally and on CircleCI before reverting back to not checking for `WASMFS`.

Reply:
I can't reproduce that even when I apply the diff from this PR and then remove the `FORCE_FILESYSTEM` on this test. So that is odd. Maybe push a commit to remove it here and we should see it fail on CI? I can look at the log then.